extension/src/setup.test.ts

```diff
@@ -618,3 +620,102 @@ describe('setup', () => {
     expect(mockedInitialize).not.toHaveBeenCalled()
   })
 })
+
+describe('setupWithGlobalRecheck', () => {
+  const extension = {
+    getAvailable: mockedGetAvailable,
+    getCliVersion: mockedGetCliVersion,
+    getRoots: mockedGetRoots,
+    hasRoots: mockedHasRoots,
+    initialize: mockedInitialize,
+    isPythonExtensionUsed: mockedIsPythonExtensionUsed,
+    resetMembers: mockedResetMembers,
+    setAvailable: mockedSetAvailable,
+    setCliCompatible: mockedSetCliCompatible,
+    setRoots: mockedSetRoots,
+    setupWorkspace: mockedSetupWorkspace,
+    unsetPythonBinPath: mockedUnsetPythonBinPath
+  }
+
+  it('should periodically check to see if the CLI is available globally if the extension fails to initialize', async () => {
```

Comment:
Jest's fake timers do not work with recursive calls to setTimeout 🤸🏻 (spent a couple of hours on this).